cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -103,8 +103,12 @@ private void deleteLocation(NamespaceId namespaceId) throws IOException {
     Location namespaceHome = namespacedLocationFactory.get(namespaceId.toId());
     try {
       if (hasCustomLocation(namespaceQueryAdmin.get(namespaceId.toId()))) {
-        LOG.debug("Custom location mapping %s was found while deleting namespace %s. Skipping location delete.",
-                  namespaceHome, namespaceId);
+        LOG.debug("Custom location mapping %s was found while deleting namespace %s. Deleting all data inside it but" +
+                    "skipping namespace home directory delete.", namespaceHome, namespaceId);
+        // delete everything inside the namespace home but not the namespace home as its user owned directory
+        for (Location childLocation : namespaceHome.list()) {
+          childLocation.delete(true);
```

Comment:
we should not delete every location like this. What data is left behind? There should be APIs to delete that individual data. Doing this might:
1. delete things that CDAP doesn't even know about
2. hide some important missing cleanup.